engineapi/proxy_snapshot.go

```diff
@@ -30,6 +32,14 @@ func (p *Proxy) SnapshotGet(e *longhorn.Engine, name string) (snapshot *longhorn
 	return recv[name], nil
 }
 
+func (p *Proxy) SnapshotRevert(e *longhorn.Engine, name string) (err error) {
+	if name == VolumeHeadName {
+		return errors.Errorf("invalid operation: cannot revert to %v", VolumeHeadName)
+	}
```

Comment:
NIT: Should the gRPC Client handle the parameter checks?

Reply:
Addressed.